init.zsh

```diff
@@ -42,7 +47,11 @@
   zstyle ':completion:*' format '%F{yellow}-- %d --%f'
   zstyle ':completion:*' group-name ''
   zstyle ':completion:*' verbose yes
-  zstyle ':completion:*' matcher-list 'm:{a-zA-Z}={A-Za-z}' '+r:|?=**'
+  if zstyle -t ':zim:completion' case-sensitive; then
+    zstyle ':completion:*' matcher-list '' 'm:{a-zA-Z}={A-Za-z}' '+r:|?=**'
```

Comment:
What about just having:
```suggestion
    zstyle ':completion:*' matcher-list 'r:|?=**'
```
here?

Reply:
oof, not what I wanted. With default case insensitive setup, it only shows matches for the middle of a word if none match anchored to the beginning. Your suggestion shows middle and beginning matches always.